forte/pipeline.py

```diff
@@ -120,6 +120,14 @@ def __init__(self, resource: Optional[Resources] = None):
             self.resource = resource
 
         self.initialized: bool = False
+        self._check_type_consistency: bool = False
+
+    def enforce_consistency(self):
+        r"""This function enforces a type consistency checking for
+        all the pipeline components.
```

Comment:
Please explain what does type consistency check is. The users won't understand what is happening here. You can use an example to explain this.